api/lib/block-listeners/base-node-subscription.js

```diff
@@ -124,9 +139,92 @@ class BaseNodeSubscription {
       this.store.update({
         validators: validators
       })
+      const storeValidators = Object.values(this.store.validators)
+      if (storeValidators.length > 0) {
+        // now that we have validators in store start polling for validator profiles
+        this.pollForValidatorsProfiles(storeValidators)
+      }
+    })
+  }
+
+  async getValidatorsProfilesFromDB(allValidatorsAddresses) {
+    const allValidatorsProfiles = await this.db.getAllValidatorsProfiles(
+      allValidatorsAddresses,
+      this.network.id
+    )
+    return keyBy(allValidatorsProfiles, `operator_address`)
+  }
+
+  async getAllValidatorsFeed(
+    validators,
+    allValidatorsAddresses,
+    networkList,
+    dataSource
+  ) {
+    const allValidatorsFeed = await this.db.getAccountsNotifications(
+      allValidatorsAddresses,
+      this.network.id
+    )
+    return validators.map((validator) => {
+      const validatorFeed = allValidatorsFeed.filter(
+        ({ resourceId }) => resourceId === validator.operatorAddress
+      )
+      return {
+        ...validator,
+        feed:
+          validatorFeed && Array.isArray(validatorFeed)
+            ? validatorFeed.map((notification) =>
+                dataSource.reducers.notificationReducer(
+                  notification,
+                  networkList
+                )
+              )
+            : []
+      }
     })
   }
 
+  async getAllValidatorsProfiles(validators, dataSource) {
```

Comment:
why is it always `getAll` instead of just `getValidatorProfiles`?

Reply:
because it conceptually comes from `getAllValidators`

Reply:
let's rename those functions. the `all` is confusing and doesn't add value

Reply:
so rename all the `getAllValidators` to just `getValidators`? OK :ok_hand: